beacon_node/network/src/sync/network_context.rs

```diff
@@ -406,6 +381,38 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
         Ok(id)
     }
 
+    fn make_columns_by_range_requests(
+        &self,
+        epoch: Epoch,
+        request: BlocksByRangeRequest,
+        custody_indexes: &Vec<ColumnIndex>,
+    ) -> Result<HashMap<PeerId, DataColumnsByRangeRequest>, RpcRequestSendError> {
+        let mut peer_id_to_request_map = HashMap::new();
+
+        for column_index in custody_indexes {
+            let Some(custody_peer) = self.get_random_custodial_peer(epoch, *column_index) else {
```

Comment:
There's potentially an issue with the peer selection approach here - as peer selection was previously done one layer up:
- [`request_batches`](https://github.com/sigp/lighthouse/blob/022b848cfb86e0e949aaddb0429a2b6206b1c9ea/beacon_node/network/src/sync/range_sync/chain.rs#L987-L999): here we use a list of "idle peers" and randomly select one
- [`retry_batch_download`](https://github.com/sigp/lighthouse/blob/022b848cfb86e0e949aaddb0429a2b6206b1c9ea/beacon_node/network/src/sync/range_sync/chain.rs#L864-L875): here we sort peers prioritizing unrelated peers with less active requests

Are these selection rules important? I've ran into some issues with concurrent requests to the same peer with the approach on this branch, and need to investigate further.

Reply:
Checked with Age, and it should be fine to send requests to peers that aren't "idle".

Reply:
The peer selection logic here isn't ideal - we should probably avoid retrying peers that failed to return a response. Perhaps we could pass the sorted peer list into this method?

Reply:
Added a comment for this, probably requires some thoughts and refactoring as we don't currently track `data_column_by_range` peers in `BatchState`.

Reply:
Note to self: we might also need to apply some of these changes to backfill sync.